components/Typography/src/MDCFontScaler.h

```diff
@@ -0,0 +1,60 @@
+#import <UIKit/UIKit.h>
+
+// Material's text styles, which are similar, but not quite equivalent to Apple's UIFontTextStyle.
+extern NSString * _Nonnull MaterialTextStyleHeadline1;
+extern NSString * _Nonnull MaterialTextStyleHeadline2;
+extern NSString * _Nonnull MaterialTextStyleHeadline3;
+extern NSString * _Nonnull MaterialTextStyleHeadline4;
+extern NSString * _Nonnull MaterialTextStyleHeadline5;
+extern NSString * _Nonnull MaterialTextStyleHeadline6;
+extern NSString * _Nonnull MaterialTextStyleSubtitle1;
+extern NSString * _Nonnull MaterialTextStyleSubtitle2;
+extern NSString * _Nonnull MaterialTextStyleBody1;
+extern NSString * _Nonnull MaterialTextStyleBody2;
+extern NSString * _Nonnull MaterialTextStyleButton;
+extern NSString * _Nonnull MaterialTextStyleCaption;
+extern NSString * _Nonnull MaterialTextStyleOverline;
+
+
+/**
+ MDCFontScaler is used to attach a scaling curve to a UIFont via an associated object on that
+ font instance.
+
+ This interface is similar to UIFontMetrics, but the font returned from MDCFontScaler do *not*
+ automatically adjust when the device's text size / content size category is changed.
+ */
+@interface MDCFontScaler : NSObject
+
+
+/**
+ Initializes a font scaler object for the specified text style.
+
+ @param textStyle The style that will be used to determine the scaling curver associated with the
+   returned font.  For example, MaterialTextStyleBody1.
+ @return An initialized font scaler object.
+ */
+- (nonnull instancetype)initForMaterialTextStyle:(nonnull NSString *)textStyle;
+
+/**
+ Creates and returns a font scaler object for the specified text style.
+
+ @param textStyle The style that will be used to determine the scaling curver associated with the
+ returned font.  For example, MaterialTextStyleBody1.
+ @return An initialized font scaler object.
+ */
++ (nonnull instancetype)scalerForMaterialTextStyle:(nonnull NSString *)textStyle;
```

Comment:
Do we expect clients to create and retain font scalers for each text style they care about?

I ask because looking at the implementation I wager this API could be just as effectively accomplished as a set of static methods:


```objc
// This:
   fontScaler = [[MDCFontScaler alloc] initForMaterialTextStyle:MaterialTextStyleHeadline2];
  _headline2 = [fontScaler scalableFontWithFont:_headline2];

// Becomes this:
  _headline2 = [MDCFontScaler scalableFontWithFont:_headline2 textStyle:MaterialTextStyleHeadline2];
```

Reply:
This interface is intentionally similar to UIFontMetrics.

Reply:
Got it. Thanks!

https://developer.apple.com/documentation/uikit/uifontmetrics